rpc/amqp-rpc.go

```diff
@@ -239,14 +259,16 @@ type RPCResponse struct {
 	Error     RPCError `json:"error,omitempty"`
 }
 
-// Start starts the AMQP-RPC server running in a separate thread.
-// There is currently no Stop() method.
-func (rpc *AmqpRPCServer) Start() (err error) {
+// Start starts the AMQP-RPC server running in a separate thread. A channel, finished,
+// is returned so that the caller can detect when the for/range messages loop has
+// broke in the spawned thread so we can cleanly exit.
+func (rpc *AmqpRPCServer) Start() (finished chan bool, err error) {
```

Comment:
I think instead this should be a blocking operation like every other server in Go.

We lose the ability to return errors that occured inside the goroutine spawned underneath and everyone has to dedicate a goroutine to reading from the channel anyway.